lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1745,6 +1951,14 @@ func (d *zfs) UnmountVolume(vol Volume, keepBlockDev bool, op *operations.Operat
 
 		d.logger.Debug("Unmounted ZFS dataset", logger.Ctx{"volName": vol.name, "dev": dataset, "path": mountPath})
 		ourUnmount = true
+
+		// If vol is a zvol, also deactivate it.
+		if d.isBlockBacking(vol) && !keepBlockDev {
+			ourUnmount, err = deactivate()
+			if err != nil {
+				return ourUnmount, err
```

Comment:
Can you just return `false` here to avoid indicating that the value returned from `deactivate` is important even if there is an error returned.